### PR TITLE
Add CI checks

### PR DESCRIPTION
**Pull request by Neilpang**
FreeBSD, NetBSD, OpenBSD and DragonflyBSD

### CODE REVIEW

File: .github/workflows/DragonflyBSD.yml [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    - uses: vmactions/dragonflybsd-vm@v0
+      with:
+        prepare: |
+          pkg install -y   mbedtls cmake
+        usesh: true
+        run: |
+          mkdir build
+          cd build
+          cmake ..
+          make && make verify
```
Comment (azat) [inline .github/workflows/DragonflyBSD.yml:42]: It is better to split building and testing:

### building

```sh
mkdir -p build
cd build
cmake ..
cmake --build .
```

### testing

```sh
JOBS=1
export CTEST_PARALLEL_LEVEL=$JOBS
export CTEST_OUTPUT_ON_FAILURE=1
cmake --build . --target verify
```

You can take a look at macos example on CI - https://github.com/libevent/libevent/blob/master/.github/workflows/build.yml#L573-L665

It is OK, to not have extended build matrix for now I guess, so building everything w/o any options will be enough

**Review by azat — APPROVED** [review]
Great stuff! Please address the comments and this will be ready to go.

**Comment by Neilpang** [comment]
@azat 

here is my first try to add freebsd CI jobs.

https://github.com/Neilpang/libevent/pull/1/files

Can you please take a look and let me know if this is what you need?
When all the jobs pass, I will send another PR to your repo.

Thanks

**Comment by Neilpang** [comment]
The new PR:  https://github.com/libevent/libevent/pull/1319

**Pull request closed without merging**